src/converters.ts

```diff
@@ -0,0 +1,228 @@
+import { RelationshipDirection } from "@jupiterone/jupiter-managed-integration-sdk";
+import {
+  SNYK_CODEREPO_ENTITY_TYPE,
+  SNYK_CODEREPO_FINDING_RELATIONSHIP_TYPE,
+  SNYK_CVE_ENTITY_TYPE,
+  SNYK_CWE_ENTITY_TYPE,
+  SNYK_FINDING_CVE_RELATIONSHIP_TYPE,
+  SNYK_FINDING_CWE_RELATIONSHIP_TYPE,
+  SNYK_FINDING_ENTITY_TYPE,
+  SNYK_SERVICE_CODEREPO_RELATIONSHIP_TYPE,
+} from "./constants";
+import {
+  CodeRepoEntity,
+  CodeRepoFindingRelationship,
+  CVEEntity,
+  CWEEntity,
+  FindingCWERelationship,
+  FindingEntity,
+  FindingVulnerabilityRelationship,
+  ServiceCodeRepoRelationship,
+  ServiceEntity,
+} from "./types";
+
+const cveLink = "https://nvd.nist.gov/vuln/detail/";
+
+function getTime(time: Date | string): number {
+  return new Date(time).getTime();
+}
+
+export interface Vulnerability {
+  id: string;
+  url: string;
+  title: string;
+  type: string;
+  description: string;
+  from: string[];
+  package: string;
+  version: string;
+  severity: string;
+  language: string;
+  packageManager: string;
+  publicationTime: Date;
+  disclosureTime: Date;
+  isUpgradable: string;
+  isPatchable: string;
+  identifiers: Identifier;
+  cvssScore: number;
+  patches: Patch[];
+  upgradePath: string[];
+}
+
+export interface Identifier {
+  CVE: string[];
+  CWE: string[];
+}
+
+export interface Patch {
+  id: string;
+  urls: string[];
+  version: string;
+  comments: string[];
+  modificationTime: Date;
+}
+
+export interface Project {
+  name: string;
+  id: string;
+  created: Date;
+  origin: string;
+  totalDependencies: number;
+  issueCountsBySeverity: IssueCount;
+}
+
+export interface IssueCount {
+  low: number;
+  medium: number;
+  high: number;
+}
+
+export function toCodeRepoEntity(project: Project): CodeRepoEntity {
+  return {
+    _class: "CodeRepo",
+    _key: `snyk-project-${project.name}`,
+    _type: SNYK_CODEREPO_ENTITY_TYPE,
+    name: project.name,
+    id: project.id,
+    created: getTime(project.created),
```

Comment:
We're going for a pattern of ending timestamps with `On`.

```suggestion
    createdOn: getTime(project.created),
```

Reply:
changed